commerce-notification-test/src/testIntegration/java/com/liferay/commerce/notification/test/CommerceOrderStatusNotificationTest.java

```diff
@@ -0,0 +1,257 @@
+
+		CommerceTestUtil.checkoutOrder(_commerceOrder);
+
+		// Notifications are asynchronous, give time to send
+
+		Thread.sleep(1000);
+
+		int commerceNotificationQueueEntriesCount =
+			_commerceNotificationQueueEntryLocalService.
+				getCommerceNotificationQueueEntriesCount(_group.getGroupId());
+
+		Assert.assertEquals(1, commerceNotificationQueueEntriesCount);
+
+		_commerceOrderLocalService.updateOrderStatus(
+			_commerceOrder.getCommerceOrderId(),
+			CommerceOrderConstants.ORDER_STATUS_AWAITING_SHIPMENT);
+
+		// Notifications are asynchronous, give time to send
+
+		Thread.sleep(1000);
+
+		commerceNotificationQueueEntriesCount =
+			_commerceNotificationQueueEntryLocalService.
+				getCommerceNotificationQueueEntriesCount(_group.getGroupId());
+
+		Assert.assertEquals(2, commerceNotificationQueueEntriesCount);
+
+		_commerceOrderLocalService.updateOrderStatus(
+			_commerceOrder.getCommerceOrderId(),
+			CommerceOrderConstants.ORDER_STATUS_SHIPPED);
+
+		// Notifications are asynchronous, give time to send
+
+		Thread.sleep(1000);
+
+		commerceNotificationQueueEntriesCount =
+			_commerceNotificationQueueEntryLocalService.
+				getCommerceNotificationQueueEntriesCount(_group.getGroupId());
+
+		Assert.assertEquals(3, commerceNotificationQueueEntriesCount);
+
+		_commerceOrderLocalService.updateOrderStatus(
+			_commerceOrder.getCommerceOrderId(),
+			CommerceOrderConstants.ORDER_STATUS_COMPLETED);
+
+		// Notifications are asynchronous, give time to send
+
+		Thread.sleep(1000);
+
+		commerceNotificationQueueEntriesCount =
+			_commerceNotificationQueueEntryLocalService.
+				getCommerceNotificationQueueEntriesCount(_group.getGroupId());
+
+		Assert.assertEquals(4, commerceNotificationQueueEntriesCount);
+	}
+
+	private void _addNotificationTemplates() throws PortalException {
+		_notificationTemplateList = new ArrayList<>(4);
+
+		CommerceNotificationTemplate commerceNotificationTemplate =
+			CommerceNotificationTestUtil.addNotificationTemplate(
+				"[%ORDER_CREATOR%]",
+				CommerceOrderConstants.ORDER_NOTIFICATION_PLACED,
+				_serviceContext);
+
+		_notificationTemplateList.add(commerceNotificationTemplate);
+
+		commerceNotificationTemplate =
+			CommerceNotificationTestUtil.addNotificationTemplate(
+				"[%ORDER_CREATOR%]",
+				CommerceOrderConstants.ORDER_NOTIFICATION_AWAITING_SHIPMENT,
+				_serviceContext);
+
+		_notificationTemplateList.add(commerceNotificationTemplate);
+
+		commerceNotificationTemplate =
+			CommerceNotificationTestUtil.addNotificationTemplate(
+				"[%ORDER_CREATOR%]",
+				CommerceOrderConstants.ORDER_NOTIFICATION_SHIPPED,
+				_serviceContext);
+
+		_notificationTemplateList.add(commerceNotificationTemplate);
+
+		commerceNotificationTemplate =
+			CommerceNotificationTestUtil.addNotificationTemplate(
+				"[%ORDER_CREATOR%]",
+				CommerceOrderConstants.ORDER_NOTIFICATION_COMPLETED,
+				_serviceContext);
+
+		_notificationTemplateList.add(commerceNotificationTemplate);
```

Comment:
This could probably be a loop, yeah? Either with an array of constants that already exists, or define an array of the statuses we care about, and then loop over that.